scrapy/templates/project/module/middlewares.py.tmpl

```diff
@@ -5,22 +5,34 @@
 # See documentation in:
 # http://doc.scrapy.org/en/latest/topics/spider-middleware.html
 
+from scrapy import signals
+
 
 class ${ProjectName}SpiderMiddleware(object):
+    # If any of methods bellow is not defined, in practice, the middleware
+    # manager only add the methods that the middleware has defined.
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        # This method is used by Scrapy to create your spiders.
+        s = cls(crawler.stats)
+        crawler.signals.connect(s.spider_opened, signal=signals.spider_opened)
```

Comment:
If I do this, looks good?

    def spider_opened(self, spider):
        spider.logger.info('Spider opened: %s' % spider.name)